### ISSUE TITLE
[css-fonts][mediaqueries-5] Add prefers-bold-text user preference media feature

### ISSUE THREAD

**Issue by gregoryd** [w3c/csswg-drafts#14477]
### Proposal Overview
Operating systems such as iOS and Android have long provided a system-level "Bold Text" toggle to aid readability for users with low vision, astigmatism, or presbyopia. However, web developers currently lack a standardized way to detect this preference and adapt typography accordingly.

We propose introducing a new user preference media query, prefers-bold-text, into the Media Queries Level 5 specification under User Preference Media Features.

### Detailed Specification & Explainer
For full details of the proposal, please see the [prefers-bold-text Explainer](https://github.com/gregoryd/prefers-bold-text/blob/main/prefers-bold-text%20Explainer.md).

**Comment by svgeesus** [comment]
Nicely written explainer. I particularly liked that for variable fonts, you suggest that Grade (`GRAD`) could be adjusted, as well as weight itself (`wght`).

Are we sure that the media query should be strictly boolean, or might this be expanded in the future? Your [suggested keywords](https://github.com/gregoryd/prefers-bold-text/blob/main/prefers-bold-text%20Explainer.md#syntax) of `no-preference` and `bold` give room to expand, good choice, but the surrounding text does make it seem like this is a boolean.

**Comment by gregoryd** [comment]
@svgeesus Thanks for the review! You're correct - the choice of keywords (no-preference and bold) was intentionally designed as an enum precisely so the feature can be expanded in the future. However, at this point, we don't have any concrete use cases for additional values.

**Comment by svgeesus** [comment]
The explainer could also make clearer the difference between adjusting `wght` and `GRAD`. [For example](https://fonts.google.com/knowledge/glossary/grade_axis)

> “Grade” (GRAD in CSS) is an [axis](https://fonts.google.com/knowledge/glossary/axis_in_variable_fonts) found in some [variable fonts](https://fonts.google.com/knowledge/glossary/variable_fonts) that can be used to alter [stroke](https://fonts.google.com/knowledge/glossary/stroke) thicknesses (or other forms) without affecting the [type](https://fonts.google.com/knowledge/glossary/type)’s overall [width](https://fonts.google.com/knowledge/glossary/width), inter-letter spacing, or [kerning](https://fonts.google.com/knowledge/glossary/kerning_kerning_pairs)—unlike altering [weight](https://fonts.google.com/knowledge/glossary/weight). This means there are no changes to line breaks or page layout.

**Comment by gregoryd** [comment]
I updated the Variable Fonts use case to highlight this.

**Comment by vitorroriz** [comment]
We should probably consider the arguments for preferring a text-scale meta tag rather than the CSS solution here as well: https://github.com/w3c/csswg-drafts/blob/main/css-env-1/explainers/meta-text-scale.md

**Comment by JoshTumath** [comment]
> We should probably consider the arguments for preferring a text-scale meta tag rather than the CSS solution here as well: https://github.com/w3c/csswg-drafts/blob/main/css-env-1/explainers/meta-text-scale.md

A meta tag was necessary in the end for text scaling because we needed a way to make sure relative units in media queries were affected by text scaling. We had to do that by changing the initial font size. If there wasn't a back compat concern, text scaling would be supported by default and we wouldn't have a `<meta text-scale>` tag.

I think it's a different situation with supporting bold text. Any solution can be implemented entirely in CSS.

**Comment by Crissov** [comment]
Is this really just about font weight? 
I would assume that users enabling this setting in browsers or operating systems would welcome **thicker strokes** in general, not just in glyphs. 

Therefore, I’d suggest to name this something like `prefers-stroke-width: thick[er] | medium |thin[ner]`.

PS: Reading just the issue title, at first I thought it might ask for a user preference between using italic and bold (or other font alterations) for emphasis.

**Comment by JoshTumath** [comment]
> Therefore, I’d suggest to name this something like `prefers-stroke-width: thick[er] | medium |thin[ner]`.

The user setting for this in different OSs is always described as 'make text bolder' so I would prefer keeping it using that language. I think we would confuse authors if the name of the preference doesn't match how users refer to it.
